contracts/modules/token/TokenController.sol

```diff
@@ -310,13 +311,68 @@ contract TokenController is ITokenController, LockHandler, MasterAwareV2 {
     }
   }
 
-  /**
-  * @dev Returns tokens locked for a specified address for a
-  *    specified reason
-  *
-  * @param _of The address whose tokens are locked
-  * @param _reason The reason to query the lock tokens for
-  */
+  /// @notice Withdraws NXM from the Nexus platform based on specified options.
+  /// @dev    Ensure the NXM is available and not locked before withdrawal. Only set flags in `WithdrawNxmOptions` for
+  ///         withdrawable NXM. Reverts if some of the NXM being withdrawn is locked or unavailable.
+  /// @param stakingPoolDeposit  Details for withdrawing from staking pools.
+  /// @param coverNotes          Details for withdrawing V1 cover notes.
+  /// @param batchSize           The maximum number of iterations to avoid unbounded loops when withdrawing
+  ///                            governance and/or assessment rewards.
+  /// @param withdrawOptions     Options specifying which NXM types to withdraw, set flags to true to include specific
+  ///                            withdrawable NXM.
+  function withdrawNXM(
+    StakingPoolDeposit calldata stakingPoolDeposit,
+    V1CoverNotes calldata coverNotes,
+    uint batchSize,
+    WithdrawNxmOptions calldata withdrawOptions
+  ) external whenNotPaused {
```

Comment:
**withdrawNXM** is not added to **ITokenController** because it's pragma (`>=0.5.0`) does not yet support structs as arguments (still experimental)